.github/workflows/pmm2-testsuite.yml

```diff
@@ -127,7 +131,39 @@ jobs:
           sleep 30
 
       - name: Setup PMM2-Client
-        run: sudo -E bash -x ./pmm-tests/pmm2-client-setup.sh --pmm_server_ip 127.0.0.1 --client_version ${{ env.CLIENT_VERSION }} --admin_password admin --use_metrics_mode no
+        run: |
+          echo "install local PMM Client..."
+          if [[ -z "${{ env.CLIENT_VERSION }}" || "${{ env.CLIENT_VERSION }}" == "dev-latest" ]]; then
+            sudo percona-release enable-only original experimental
+            sudo apt-get update
+            sudo apt-get -y install pmm2-client
+          fi
+          if [[ "${{ env.CLIENT_VERSION }}" == "pmm2-rc" ]]; then
+            sudo percona-release enable-only original testing
+            sudo apt-get update
+            sudo apt-get -y install pmm2-client
+          fi
+          if [[ "${{ env.CLIENT_VERSION }}" == "pmm2-latest" ]]; then
+            sudo apt-get -y install pmm2-client
+            sudo apt-get -y update
+          fi
+          if [[ "${{ env.CLIENT_VERSION }}" == http* || "${{ env.CLIENT_VERSION }}" == 2* ]]; then
+            curl -O https://raw.githubusercontent.com/Percona-QA/package-testing/PMM-7-tarball-improve/scripts/pmm2_client_install_tarball.sh
+            sudo bash -x pmm2_client_install_tarball.sh -v ${{ env.CLIENT_VERSION }}
+            echo 'Setup local pmm-agent...'
+            sudo pmm-agent setup --force --config-file=/usr/local/percona/pmm2/config/pmm-agent.yaml --server-address=127.0.0.1:443 --server-insecure-tls --server-username=admin --server-password=admin
+            sudo pmm-agent --config-file=/usr/local/percona/pmm2/config/pmm-agent.yaml > pmm-agent.log 2>&1 &
+            sleep 2
+          else
+            echo 'Setup installed PMM Client package...'
+            sudo pmm-admin config --force --server-insecure-tls --metrics-mode=auto --server-url=https://admin:admin@127.0.0.1:443
+            sudo usermod -a -G pmm-agent $USER
```

Comment:
I am not able to understand what case would fall on this condition?

Reply:
this condition is for package install - versions "empty", "dev-latest", "pmm2-rc" and  "pmm2-latest", this is the same way as in original script: https://github.com/percona/pmm-qa/blob/main/pmm-tests/pmm2-client-setup.sh#L101
I didn't find better structure to extract given closure without big duplications of code